### PR TITLE
t: basic tests for Rose Bush

### PR DESCRIPTION
**Pull request by matthewrmshin**
1st step for #1010. More to come.

### CODE REVIEW

**Comment by matthewrmshin** [comment]
@arjclark @kaday please review.

File: t/rose-bush/00-basic.t [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+    0<'/dev/null' 1>'rose-bush.out' 2>'rose-bush.err' &
+ROSE_BUSH_PID=$!
+T_INIT="$(date '+%s')"
+while ! port_is_busy "${PORT}" && (($(date '+%s') < ${T_INIT} + 60)); do
+    sleep 1
+done
+if port_is_busy "${PORT}"; then
+    pass "${TEST_KEY}"
+else
+    fail "${TEST_KEY}"
```
Comment (arjclark) [inline t/rose-bush/00-basic.t]: missing kill `$ROSE_BUSH_PID`?

Comment (matthewrmshin) [inline t/rose-bush/00-basic.t]: Done.

**Comment by arjclark** [comment]
Looks ok to me. Tests working in my environment and extra kill invocation has been added. Over to @kaday

**Comment by matthewrmshin** [comment]
Branch re-based + improved tests concurrency.

**Comment by kaday** [comment]
Looks okay to me, tests all pass in my environment.

**Pull request merged**